ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -679,17 +677,15 @@ private boolean performOperation(AsynchronousFileChannel fileChannel, ByteBuffer
      */
     private AsynchronousFileChannel createFileChannel() {
       try {
-        File dir = new File(healthcheckDir);
-        if (!dir.exists() && !dir.mkdir()) {
-          throw new Error(String.format("%s directory wasn't created", healthcheckDir));
-        }
-        Path path = Paths.get(healthcheckDir, "temp");
+        checkMountPathAccessible();
+        Path path = Paths.get(disk.getMountPath(), "temp");
         AsynchronousFileChannel fileChannel =
             AsynchronousFileChannel.open(path, StandardOpenOption.WRITE, StandardOpenOption.READ,
                 StandardOpenOption.SYNC, StandardOpenOption.CREATE, StandardOpenOption.DELETE_ON_CLOSE);
         return fileChannel;
       } catch (Exception e) {
-        diskHealthStatus = DiskHealthStatus.CREATE_EXCEPTION;
+        diskHealthStatus =
+            e instanceof StoreException ? DiskHealthStatus.MOUNT_NOT_ACCESSIBLE : DiskHealthStatus.CREATE_EXCEPTION;
```

Comment:
Instead of just checking if e is an instance of StoreException, check specifically that the message or errcode from L539 matches your expectation.

Reply:
And add a new catch blocks to catch StoreException

Reply:
You have a good point, good practice to differentiate the exceptions and be exact about it. Fixed